### DISCUSSION TITLE
timezone and UTC

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5110] [Q&A Core] [answered]
Just from curiosity.  
If I use the different timezone in e107 than UTC , Wordpress will complain about it:  "PHP default timezone was changed after WordPress loading by a date_default_timezone_set() function call. This interferes with correct calculations of dates and times."
Timezone setting is correct -  +1 hour  -  in WP settings. 

Could somebody explain me why e107 can change default timezone and WP doesn't allow it? 
Maybe I am doing it wrong in e107 - I should use UTC and then set user timezone to Central Europe. 

I tried to read something, but why this difference?   

I found this on that topic from WP side:  https://make.wordpress.org/core/2019/09/23/date-time-improvements-wp-5-3/
but nothing from e107.

Any light on this? 
Thanks

**Comment by Deltik** [comment, marked as answer]
It looks like WordPress introduced their own framework for datetime handling, which requires `date_default_timezone_set()` not to be called, but e107 does this sometimes, particularly [in `init_session()`](https://github.com/e107inc/e107/blob/9285c24e86d6a61cf11f8bd56f89004a98e72c9e/class2.php#L1563-L1563).

Perhaps you can work around this by starting up the e107 session first and then loading WordPress.  Or if you don't need an e107 user session, set the global variable `$_E107['no_session']` to a truthy value to disable the e107 user session.

**Reply by Jimmi08** [reply]
Thank you.  Is there a difference between $_E107['no_session'] and shutdown() from the session handler?

**Reply by Deltik** [reply]
Supposedly, `$_E107['no_session']` doesn't even start the session at all.  This is not a code path that we regularly test.
